modules/default/weather/weatherobject.js

```diff
@@ -146,6 +146,20 @@ class WeatherObject {
 		this.sunrise = moment(times.sunrise, "X");
 		this.sunset = moment(times.sunset, "X");
 	}
+
+	/**
+	 * Clone to simple object to prevent mutating and deprecated legacy library.
+	 *
+	 * @returns {object} simple object cloned.
+	 */
+	simpleClone() {
```

Comment:
since there are so many js features used in this method (which might confuse beginners) would be nice to have an example of what this does in the jsdoc.

Reply:
I'm not sure a good example could be possible because this feature is only for making a safe-immutable payload of notification.
Anyway, I added some description more.

Reply:
@eouia @rejas   all the default modules send notifications of their data so that others might compose new apps.. 
BUT since the new weather module combined current weather and forecast, it (the new module) does NOT broadcast the forecast info when running in forecast mode.

Reply:
By the way, all the payload of original default modules is mutable and not safe from modifying by recipient modules I think that could be a risk in some cases.

Reply:
> By the way, all the payload of original default modules is mutable and not safe from modifying by recipient modules I think that could be a risk in some cases.

so thats why you added this method?

Reply:
Also: I am wondering if there isnt a lodash method that would do the same thing?

Reply:
> @eouia @rejas all the default modules send notifications of their data so that others might compose new apps.. BUT since the new weather module combined current weather and forecast, it (the new module) does NOT broadcast the forecast info when running in forecast mode.

yes, it only published the CURRENTWEATHER_TYPE, so that the compliments module can react to it ( I think I added that back in the days) Maybe its a good idea to remove that call and broadcast only the whole weather object (instead of just the CURRENTWEATHER_TYPE)?

Reply:
moment object is not immutable and unsafe.

For example; In case of ‘weatherObject.sunrise’ is handed to other module as payload directly, the 3rd party module can manipulate the object;

```js
/* In other module’s receivedNotification function */
let afterSunRise = payload.sunrise.add(30, ‘minute’)
```
But by this execution, the original `this.weatherProvider.currentWeatherObject.sunrise` Will be changed. Usually this is a bad side-effect unintentional.

Reply:
I dont mind 3rd party libraries like lodash, sicne they are well tested and we already use it in here. But your code your choice :-)